sdk/keyvault/azure-security-keyvault-jca/README.md

```diff
@@ -190,6 +365,143 @@ Please replace `${KEY_VAULT}` with your key vault name and replace `${MANAGED_ID
+
+3. Create a key vault
+```shell
+az keyvault create --name $KEYVAULT_NAME --resource-group $RESOURCE_GROUP_NAME --location "EastUS"
+```
+
+4. Assign role to create certificates in the Key Vault.
+```shell
+# Get your user object ID (if you're using a user account)
+USER_OBJECTID=$(az ad signed-in-user show --query id -o tsv)
+
+# Or if you're using a service principal, get its object ID
+# SP_OBJECTID=$(az ad sp show --id <your-sp-id> --query id -o tsv)
+
+# Assign Key Vault Certificates Officer role
+az role assignment create \
+    --role "Key Vault Certificates Officer" \
+    --assignee $USER_OBJECTID \
+    --scope "/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP_NAME/providers/Microsoft.KeyVault/vaults/$KEYVAULT_NAME"
+```
+
+5. Get the Key Vault URL
+```shell
+KEYVAULT_URL=$(az keyvault show --name $KEYVAULT_NAME --query "properties.vaultUri" --resource-group $RESOURCE_GROUP_NAME -o tsv| tr -d '\r\n')
+echo $KEYVAULT_URL
+```
+
+6. Add a certificate to Key Vault
+```shell
+az keyvault certificate create --vault-name $KEYVAULT_NAME -n $CERT_NAME -p "$(az keyvault certificate get-default-policy)"
+```
+
+7. Create a Service Principal
+```shell
+SP_JSON=$(az ad sp create-for-rbac --name $SERVICE_PRINCIPAL_NAME)
+
+CLIENT_ID=$(echo $SP_JSON | jq -r '.appId')
+CLIENT_SECRET=$(echo $SP_JSON | jq -r '.password')
+TENANT=$(echo $SP_JSON | jq -r '.tenant')
+
+echo "CLIENT_ID:"$CLIENT_ID
+echo "CLIENT_SECRET:"$CLIENT_SECRET
+echo "TENANT:"$TENANT
+```
+Note the appId and password from the output, you'll need them later.
+
+8. Get the objectId
+```shell
+OBJECTID=$(az ad sp show --id "$CLIENT_ID" --query id -o tsv | tr -d '\r\n')
+echo $OBJECTID
+```
+
+9. Assign Roles to Service Principal:
+```shell
+# Assign Key Vault Secrets Officer role to Service Principal
+az role assignment create \
+    --role "Key Vault Secrets Officer" \
+    --assignee $OBJECTID \
+    --scope "/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP_NAME/providers/Microsoft.KeyVault/vaults/$KEYVAULT_NAME"
+
+# Assign Key Vault Certificates Officer role Service Principal
+az role assignment create \
+    --role "Key Vault Certificates Officer" \
+    --assignee $OBJECTID \
+    --scope "/subscriptions/$SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP_NAME/providers/Microsoft.KeyVault/vaults/$KEYVAULT_NAME"
+```
+
+### Sign with Jarsigner
+1. If you are using Java8, try to sign the jar using below command
+ ```bash
+ jarsigner   -keystore NONE -storetype AzureKeyVault \
+             -signedjar signerjar.jar ${PARAM_YOUR_JAR_FILE_PATH} "${CERT_NAME}" \
+             -verbose  -storepass "" \
+             -providerName AzureKeyVault \
+             -providerClass com.azure.security.keyvault.jca.KeyVaultJcaProvider \
+             -J-Dazure.keyvault.uri=${KEYVAULT_URL} \
+             -J-Dazure.keyvault.tenant-id=${TENANT} \
+             -J-Dazure.keyvault.client-id=${CLIENT_ID} \
+             -J-Dazure.keyvault.client-secret=${CLIENT_SECRET}
+ ```
+
```

Comment:
Should we add output logs or screenshots to record actions because we will forget them over time?